libtesting-ui/src/main/java/com/mapbox/navigation/testing/ui/http/MockWebServerRule.kt

```diff
@@ -0,0 +1,68 @@
+package com.mapbox.navigation.testing.ui.http
+
+import okhttp3.mockwebserver.Dispatcher
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import okhttp3.mockwebserver.RecordedRequest
+import org.junit.rules.TestWatcher
+import org.junit.runner.Description
+
+/**
+ * Creates and initializes a [MockWebServer] for each test.
```

Comment:
This is expensive. Wondering if we could do it [`@BeforeClass`](https://www.baeldung.com/junit-before-beforeclass-beforeeach-beforeall#beforeclass) instead of on every test 🤔

Reply:
We could, but then we'd need to clear the added request handlers manually in each test to keep the tests well isolated. I don't see recreating the server causing any overhead atm, but we can revisit in the future.